Makefile

```diff
@@ -295,6 +295,8 @@ VERSION := $(shell git describe --tags --always --dirty)
 endif
 export VERSION
 
+GIT_COMMIT := $(if $(SOURCE_GIT_COMMIT),$(SOURCE_GIT_COMMIT),$(shell git rev-parse HEAD))
+
```

Comment:
I don't think we need this line. If we (or a fork) want to build binaries with a specific GIT_COMMIT reference, we/they should explicitly set `GIT_COMMIT`

When `GIT_COMMIT` is left unset, we will also set `$(VERSION_PATH).gitCommit` to empty string. So we should:
1. Initialize `gitCommit` as `""` rather than `"unknown"` [here](https://github.com/operator-framework/operator-controller/blob/a6de9f94e085989a086459f2b5325a5cd27424b0/internal/shared/version/version.go#L9).
2. Put [this](https://github.com/operator-framework/operator-controller/blob/a6de9f94e085989a086459f2b5325a5cd27424b0/internal/shared/version/version.go#L32) in an if statement that only sets from the build data when `gitCommit == ""`.

Reply:
Addressed based on this feedback in the latest commit.

This approach seems better as it eliminates the need to introduce an openshift specific envvar upstream.